src/EnergyPlus/SteamBaseboardRadiator.hh

```diff
@@ -111,20 +112,36 @@ namespace SteamBaseboardRadiator {
         int BBLoadReSimIndex;
         int BBMassFlowReSimIndex;
         int BBInletTempFlowReSimIndex;
-        int HeatingCapMethod; // - Method for steam baseboard Radiator system heating capacity scaledsizing calculation (HeatingDesignCapacity,
-                              // CapacityPerFloorArea, FracOfAutosizedHeatingCapacity)
         Real64 ScaledHeatingCapacity; // -  steam baseboard Radiator system scaled maximum heating capacity {W} or scalable variable of zone HVAC
                                       // equipment, {-}, or {W/m2}
 
         // Default Constructor
         SteamBaseboardParams()
-            : EquipType(0), ZonePtr(0), SchedPtr(0), SteamInletNode(0), SteamOutletNode(0), TotSurfToDistrib(0), FluidIndex(0), ControlCompTypeNum(0),
-              CompErrIndex(0), DegOfSubcooling(0.0), Offset(0.0), SteamMassFlowRate(0.0), SteamMassFlowRateMax(0.0), SteamVolFlowRateMax(0.0),
+            : EquipType(0), DesignObjectPtr(0), ZonePtr(0), SchedPtr(0), SteamInletNode(0), SteamOutletNode(0), TotSurfToDistrib(0), FluidIndex(0), ControlCompTypeNum(0),
+              CompErrIndex(0), DegOfSubcooling(0.0), SteamMassFlowRate(0.0), SteamMassFlowRateMax(0.0), SteamVolFlowRateMax(0.0),
               SteamOutletTemp(0.0), SteamInletTemp(0.0), SteamInletEnthalpy(0.0), SteamOutletEnthalpy(0.0), SteamInletPress(0.0),
               SteamOutletPress(0.0), SteamInletQuality(0.0), SteamOutletQuality(0.0), FracRadiant(0.0), FracConvect(0.0), FracDistribPerson(0.0),
               TotPower(0.0), Power(0.0), ConvPower(0.0), RadPower(0.0), TotEnergy(0.0), Energy(0.0), ConvEnergy(0.0), RadEnergy(0.0), LoopNum(0),
               LoopSideNum(0), BranchNum(0), CompNum(0), BBLoadReSimIndex(0), BBMassFlowReSimIndex(0), BBInletTempFlowReSimIndex(0),
-              HeatingCapMethod(0), ScaledHeatingCapacity(0.0)
+             ScaledHeatingCapacity(0.0)
+        {
+        }
+    };
+
+    struct SteamBaseboardDesignData : SteamBaseboardParams
+    {
+        // Members
+        std::string designName;
+        int HeatingCapMethod;         // - Method for heating capacity scaledsizing calculation (HeatingDesignCapacity, CapacityPerFloorArea,
+        // FracOfAutosizedHeatingCapacity)
+        Real64 DesignScaledHeatingCapacity; // - scaled maximum heating capacity {W} or scalable variable of zone HVAC equipment, {-}, or {W/m2}
+        Real64 Offset;
+        Real64 FracRadiant;
+        Real64 FracDistribPerson;
+
+        // Default Constructor
+        SteamBaseboardDesignData()
+                : HeatingCapMethod(0), DesignScaledHeatingCapacity(0.0), Offset(0.0), FracRadiant(0.0), FracDistribPerson(0.0)
```

Comment:
I know this is how most default constructors are written in EnergyPlus, so I don't want you to change anything for now, but I wanted to point out that I think we should move to initializing right at the declaration.  In other words, instead of having to write this initializer list, just set heating method to 0 right where it is declared: `int HeatingCapMethod = 0`.  The compiler is quite smart, and at even just level 1 optimization, gcc will end up with the same assembly code either way.  And not that it matters, but in debug builds, the code is much more efficient if you are able to initialize the way I am suggesting.  If someone wants to correct me here, I'm open to discussion, this is just what I'm seeing on CompilerExplorer.

Reply:
Thanks for this.